sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -121,19 +126,22 @@ export function convertResponseForQueryBatch(
 
         return left - right;
       })
-      ?.map((response: GeneratedBatchQueryResponse) => ({
-        id: response.id,
-        status: response.status,
-        // hoist fields from the sub-object 'body' to this level
-        error: response.body?.error,
-        tables: response.body?.tables?.map(convertGeneratedTable)
-      }))
+      ?.map((response: GeneratedBatchQueryResponse) => convertBatchQueryResponseHelper(response))
   };
-
+  // compute status for failed or succeed or partial results
+
+  for (let i = 0; i < newResponse.results?.length; i++) {
+    if (newResponse.results[i].error && newResponse.results[i].tables) {
+      newResponse.results[i].status = "Partial";
+    } else if (newResponse.results[i].tables) {
+      newResponse.results[i].status = "Success";
+    } else {
+      newResponse.results[i].status = "Failed";
+    }
+  }
```

Comment:
If `results` is undefined, then we're doing a `i < undefined` check. This returns false, but can we make sure we always compare same types? We can also store the result we're comparing/setting so we don't have to keep chaining into it.

Maybe something like:
```suggestion
  const resultsCount = newResponse.results?.length ?? 0;
  for (let i = 0; i < resultsCount; i++) {
    const result = newResponse.results[i];
    if (result.error && result.tables) {
      result.status = "Partial";
    } else if (result.tables) {
       result.status = "Success";
    } else {
      result.status = "Failed";
    }
  }
```